File: backend/schemas/filme.py
from enum import Enum

from .base import Base

class GeneroFilme(Base):
    id: int
    nome: str

class ImagensFilme(Base):
    capa: str | None = None #storage+poster_path
    banner: str | None = None #storage+backdrop_path

class FilmeRead(Base):
    id: int
    titulo: str
    titulo_original: str
    idioma_original: str
    descricao: str | None = None
    status: str # TMDB não disponibiliza explicitamente a lista com todos os valores possíveis para Status
    data_lancamento: str | None = None
    duracao_minutos: int = 0 # 0 é o padrão da api do TMDB
    imagens: ImagensFilme
    generos: list[GeneroFilme]

File: frontend/lib/mocks/filme.ts
import { Filme, FilmeListagem } from "@/lib/types/filme";

export const filmeListagemMock: FilmeListagem = {
    id: 2001,
    titulo: "O Último Farol",
    titulo_original: "The Last Lighthouse",
    idioma_original: "en",
    descricao:
        "Uma fotógrafa retorna à ilha onde cresceu para investigar o desaparecimento de um antigo faroleiro.",
    status: "Released",
    data_lancamento: "2024-10-18",
    imagens: {
        capa: "https://images.unsplash.com/photo-1500534623283-312aade485b7",
        banner: "https://images.unsplash.com/photo-1500534314209-a25ddb2bd429",
    },
    generos_ids: [18, 9648],
};

export const filmeMock: Filme = {
    id: filmeListagemMock.id,
    titulo: filmeListagemMock.titulo,
    titulo_original: filmeListagemMock.titulo_original,
    idioma_original: filmeListagemMock.idioma_original,
    descricao: filmeListagemMock.descricao,
    status: filmeListagemMock.status,
    data_lancamento: filmeListagemMock.data_lancamento,
    imagens: filmeListagemMock.imagens,
    duracao_minutos: 118,
    generos: [
        { id: 18, nome: "Drama" },
        { id: 9648, nome: "Mistério" },
    ],
};
